drivers/timer/riscv_machine_timer.c

```diff
@@ -78,6 +78,32 @@
 
 /* the unsigned long cast limits divisions to native CPU register width */
 #define cycle_diff_t unsigned long
+#define CYCLE_DIFF_MAX (~(cycle_diff_t)0)
+
+/*
+ * We have two constraints on the maximum number of cycles we can wait for.
+ *
+ * 1) sys_clock_announce() accepts at most INT32_MAX ticks.
+ *
+ * 2) The number of cycles between two reports must fit in a cycle_diff_t
+ *    variable before converting it to ticks.
+ *
+ * Then:
+ *
+ * 3) Pick the smallest between (1) and (2).
+ *
+ * 4) Take into account some room for the unavoidable IRQ servicing latency.
+ *    Let's use 3/4 of the max range.
+ *
+ * Finally let's add the LSB value to the result so to clear out a bunch of
+ * consecutive set bits coming from the original max values to produce a
+ * nicer literal for assembly generation.
+ */
+#define CYCLES_MAX_1	((uint64_t)INT32_MAX * (uint64_t)CYC_PER_TICK)
```

Comment:
Style: if it's worth splitting these out into separate macros, it's probably worth finding names for them, maybe:

```c
#define THREE_QUARTERS(x) ((x) / 2 + (x) / 4)

/* Description as above */
#define LSB_BUMP(x) ((x) + LSB_GET(x))

#define TICK_MAX ... // as for _1
#define TYPE_MAX ... // ditto _2
#define CYCLES_MAX LSB_BUMP(THREE_QUARTERS(MIN(TICK_MAX, TYPE_MAX)))
```